### DISCUSSION TITLE
Downcasting a Datum to a CustomClass Type

### DISCUSSION THREAD

**Discussion by while0x1** [OpShin/opshin#42] [Q&A] [answered]
I have the following Function
```
def getWinnersDatum(txins: List[TxInInfo],token: Token) -> Datum:
    for txi in txins:
        if txi.resolved.value.get(token.policy_id, {b"": 0}).get(token.token_name, 0) > 0:
            unionDatum = txi.resolved.datum
        if isinstance(unionDatum, SomeOutputDatum):    
            winnersDatum = unionDatum.datum
    return winnersDatum
```
 which compiles but I am not sure of how to proceed from getting the datum to making an actual comparison of the reference input datum with a custom class (the format i expect the datum to be submitted in) -If I cast the datum to a custom type the compiler throws an error :

winnersDatum = winnerList(unionDatum.datum)
                  ^
AssertionError: Signature of function Call(func=Name(id='winnerList', ctx=Load()), args=[Attribute(value=Name(id='unionDatum', ctx=Load()), attr='datum', ctx=Load())], keywords=[]) does not match arguments

**Comment by nielstron** [comment, marked as answer]
Use a Downcast by annotating the assigned variable with the desired type like this
```python
winnerdatum: winnerList = unionDatum.datum
```


Here is a another code example with explanation: https://github.com/ImperatorLang/eopsin/blob/master/examples/datum_cast.py

**Reply by while0x1** [reply]
Thanks! That worked - and the contract compiles! Now if it actually does what I think it will is a completely different story. Feeding the downcasted datum into another function which iterates over the winnerList type also compiled so hopefully on the right track.
